notify/notify.go

```diff
@@ -44,7 +44,7 @@ type ResolvedSender interface {
 // Peer represents the cluster node from where we are the sending the notification.
 type Peer interface {
 	// WaitReady waits until the node silences and notifications have settled before attempting to send a notification.
-	WaitReady()
+	WaitReady(context.Context)
```

Comment:
When accepting context, I would expect that method also returns `error`, so that client calling it knows whether method has succeded or not.

Reply:
Also added a test for the non-blocking path.